### PR TITLE
JAMES-3098 - filter properties in JMAP get mailbox method

### PR DESCRIPTION
**Pull request by remk**

### CODE REVIEW

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Mailbox.scala [hunk 1]
```diff
@@ -142,4 +144,25 @@ case class Mailbox(id: MailboxId,
...
+      CapabilityIdentifier.JAMES_SHARES -> Set("namespace", "rights")
+    )
+
+    val propertiesToHide = propertiesForCapabilities.filterNot(entry => allowedCapabilities.contains(entry._1))
+      .flatMap(_._2)
+      .toSet
+
+    requestedProperties match {
+      case None => allProperties -- propertiesToHide
+      case Some(requested) => (Set("id") ++  requested.value.map(_.toString()).toSet) -- propertiesToHide
```
Comment (rouazana) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Mailbox.scala:165]:
```suggestion
      case Some(requested) => (Set("id") ++ requested.value.map(_.toString()).toSet) -- propertiesToHide
```

**Comment by remk** [comment]
> do you miss a test where properties is defined but empty?
I already have the one where i request only the name and check that i get the id too.
But i can add it.

**Comment by rouazana** [comment]
> > do you miss a test where properties is defined but empty?
> > I already have the one where i request only the name and check that i get the id too.
> > But i can add it.

yes please it's not exactly the same case

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxGetMethodContract.scala [hunk 2]
```diff
@@ -367,6 +458,296 @@ trait MailboxGetMethodContract {
...
+    val mailboxId: String = server.getProbe(classOf[MailboxProbeImpl])
+      .createMailbox(MailboxPath.forUser(BOB, "custom"))
+      .serialize
+
+    val response: String = `given`
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(GET_ALL_MAILBOXES_REQUEST_NULL_PROPERTIES)
+      .when
+      .post
+      .`then`
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxGetMethodContract.scala:527]: Should we maintain the indent rule for given/when/then ?

**Comment by Arsnael** [comment]
```
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95] Serialize MailboxGetResponse
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95] - should succeed *** FAILED ***
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   org.opentest4j.AssertionFailedError: JSON documents are different:
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95] Different keys found in node "list[0]", missing: "list[0].id","list[0].isSubscribed","list[0].myRights","list[0].name","list[0].namespace","list[0].parentId","list[0].quotas","list[0].rights","list[0].role","list[0].sortOrder","list[0].totalEmails","list[0].totalThreads","list[0].unreadEmails","list[0].unreadThreads", expected: <{"id":"2","isSubscribed":true,"myRights":{"mayAddItems":true,"mayCreateChild":true,"mayDelete":false,"mayReadItems":false,"mayRemoveItems":false,"mayRename":true,"maySetKeywords":false,"maySetSeen":true,"maySubmit":false},"name":"inbox","namespace":"Personal","parentId":"1","quotas":{"quotaRoot":{"Message":{"max":42,"used":18},"Storage":{"used":12}},"quotaRoot2@localhost":{"Message":{"max":43,"used":14},"Storage":{"used":19}}},"rights":{"alice":[r, w],"bob":[e, l]},"role":"inbox","sortOrder":10,"totalEmails":1234,"totalThreads":58,"unreadEmails":123,"unreadThreads":22}> but was: <{}>
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at net.javacrumbs.jsonunit.core.internal.ExceptionUtils.createException(ExceptionUtils.java:48)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at net.javacrumbs.jsonunit.core.internal.Diff.failIfDifferent(Diff.java:613)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at net.javacrumbs.jsonunit.assertj.JsonAssert.isEqualTo(JsonAssert.java:106)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at org.apache.james.jmap.json.MailboxGetSerializationTest.$anonfun$new$10(MailboxGetSerializationTest.scala:210)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at org.scalatest.wordspec.AnyWordSpecLike$$anon$3.apply(AnyWordSpecLike.scala:1076)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   ...
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95] MailboxSerializationTest:
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95] Serialize Mailbox
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95] - should succeed  *** FAILED ***
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   org.opentest4j.AssertionFailedError: JSON documents are different:
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95] Different keys found in node "", missing: "id","isSubscribed","myRights","name","namespace","parentId","quotas","rights","role","sortOrder","totalEmails","totalThreads","unreadEmails","unreadThreads", expected: <{"id":"2","isSubscribed":true,"myRights":{"mayAddItems":true,"mayCreateChild":true,"mayDelete":false,"mayReadItems":false,"mayRemoveItems":false,"mayRename":true,"maySetKeywords":false,"maySetSeen":true,"maySubmit":false},"name":"inbox","namespace":"Personal","parentId":"1","quotas":{"quotaRoot":{"Message":{"max":42,"used":18},"Storage":{"used":12}},"quotaRoot2@localhost":{"Message":{"max":43,"used":14},"Storage":{"used":19}}},"rights":{"alice":[r, w],"bob":[e, l]},"role":"inbox","sortOrder":10,"totalEmails":1234,"totalThreads":58,"unreadEmails":123,"unreadThreads":22}> but was: <{}>
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at net.javacrumbs.jsonunit.core.internal.ExceptionUtils.createException(ExceptionUtils.java:48)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at net.javacrumbs.jsonunit.core.internal.Diff.failIfDifferent(Diff.java:613)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at net.javacrumbs.jsonunit.assertj.JsonAssert.isEqualTo(JsonAssert.java:106)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at org.apache.james.jmap.json.MailboxSerializationTest.$anonfun$new$2(MailboxSerializationTest.scala:134)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   at org.scalatest.wordspec.AnyWordSpecLike$$anon$3.apply(AnyWordSpecLike.scala:1076)
[16e6b24b9c1efbb7ba84324370b9be034ccf7e95]   ...
```
Definitely some tests to adapt... let me help you with this

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxGetMethodContract.scala (hunk 2 above)
Comment (Arsnael) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxGetMethodContract.scala]: I did indent them when fixing the errors in tests, but the question still stands I guess

**Comment by remk** [comment]
> Definitely some tests to adapt... let me help you with this

Thank you :heart_eyes:

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**